diff --git a/node_modules/@aehrc/sdc-template-extract/dist/index.cjs b/node_modules/@aehrc/sdc-template-extract/dist/index.cjs
index efbfd88..e783d76 100644
--- a/node_modules/@aehrc/sdc-template-extract/dist/index.cjs
+++ b/node_modules/@aehrc/sdc-template-extract/dist/index.cjs
@@ -1081,6 +1081,19 @@ function insertValuesToPath(entryPath, insertIndex, isNewInsert, valuePath, valu
     );
   }
 }
+function mergeArrayByIndex(target, source, options) {
+  const destination = target.slice();
+  source.forEach((item, index) => {
+    if (typeof destination[index] === "undefined") {
+      destination[index] = options.cloneUnlessOtherwiseSpecified(item, options);
+    } else if (options.isMergeableObject(item)) {
+      destination[index] = (0, import_deepmerge.default)(target[index], item, options);
+    } else if (target.indexOf(item) === -1) {
+      destination.push(item);
+    }
+  });
+  return destination;
+}
 function walkTemplateAndInsertValue(obj, entryPath, entryPathSegments, isNewInsert, valueToInsert) {
   if (!obj || typeof obj !== "object") {
     return;
@@ -1126,7 +1139,7 @@ function walkTemplateAndInsertValue(obj, entryPath, entryPathSegments, isNewInse
     return;
   }
   if (typeof existingNode === "object" && typeof valueToInsert === "object") {
-    current[finalSegment] = (0, import_deepmerge.default)(existingNode, valueToInsert);
+    current[finalSegment] = (0, import_deepmerge.default)(existingNode, valueToInsert, { arrayMerge: mergeArrayByIndex });
     return;
   }
   current[finalSegment] = valueToInsert;
diff --git a/node_modules/@aehrc/sdc-template-extract/dist/index.js b/node_modules/@aehrc/sdc-template-extract/dist/index.js
index c41ad5c..67b8acd 100644
--- a/node_modules/@aehrc/sdc-template-extract/dist/index.js
+++ b/node_modules/@aehrc/sdc-template-extract/dist/index.js
@@ -1038,6 +1038,19 @@ function insertValuesToPath(entryPath, insertIndex, isNewInsert, valuePath, valu
     );
   }
 }
+function mergeArrayByIndex(target, source, options) {
+  const destination = target.slice();
+  source.forEach((item, index) => {
+    if (typeof destination[index] === "undefined") {
+      destination[index] = options.cloneUnlessOtherwiseSpecified(item, options);
+    } else if (options.isMergeableObject(item)) {
+      destination[index] = deepmerge(target[index], item, options);
+    } else if (target.indexOf(item) === -1) {
+      destination.push(item);
+    }
+  });
+  return destination;
+}
 function walkTemplateAndInsertValue(obj, entryPath, entryPathSegments, isNewInsert, valueToInsert) {
   if (!obj || typeof obj !== "object") {
     return;
@@ -1083,7 +1096,7 @@ function walkTemplateAndInsertValue(obj, entryPath, entryPathSegments, isNewInse
     return;
   }
   if (typeof existingNode === "object" && typeof valueToInsert === "object") {
-    current[finalSegment] = deepmerge(existingNode, valueToInsert);
+    current[finalSegment] = deepmerge(existingNode, valueToInsert, { arrayMerge: mergeArrayByIndex });
     return;
   }
   current[finalSegment] = valueToInsert;
