diff --git a/node_modules/@aehrc/sdc-assemble/dist/index.cjs b/node_modules/@aehrc/sdc-assemble/dist/index.cjs
index f1178dd..a844c68 100644
--- a/node_modules/@aehrc/sdc-assemble/dist/index.cjs
+++ b/node_modules/@aehrc/sdc-assemble/dist/index.cjs
@@ -655,9 +655,9 @@ function assembleQuestionnaire(rootQuestionnaire, totalCanonicals, issues, isRoo
     if (!Array.isArray(subquestionnaires)) {
       return subquestionnaires;
     }
-    for (let subquestionnaire of subquestionnaires) {
+    for (let i = 0; i < subquestionnaires.length; i++) {
       const assembled = yield assembleQuestionnaire(
-        subquestionnaire,
+        subquestionnaires[i],
         totalCanonicals,
         issues,
         false,
@@ -668,7 +668,7 @@ function assembleQuestionnaire(rootQuestionnaire, totalCanonicals, issues, isRoo
       if (assembled.resourceType === "OperationOutcome") {
         return assembled;
       }
-      subquestionnaire = assembled;
+      subquestionnaires[i] = assembled;
     }
     const prohibitedAttributesOutcome = checkProhibitedAttributes(subquestionnaires);
     if (prohibitedAttributesOutcome) {
diff --git a/node_modules/@aehrc/sdc-assemble/dist/index.js b/node_modules/@aehrc/sdc-assemble/dist/index.js
index 701dceb..7b2f91d 100644
--- a/node_modules/@aehrc/sdc-assemble/dist/index.js
+++ b/node_modules/@aehrc/sdc-assemble/dist/index.js
@@ -619,9 +619,9 @@ function assembleQuestionnaire(rootQuestionnaire, totalCanonicals, issues, isRoo
     if (!Array.isArray(subquestionnaires)) {
       return subquestionnaires;
     }
-    for (let subquestionnaire of subquestionnaires) {
+    for (let i = 0; i < subquestionnaires.length; i++) {
       const assembled = yield assembleQuestionnaire(
-        subquestionnaire,
+        subquestionnaires[i],
         totalCanonicals,
         issues,
         false,
@@ -632,7 +632,7 @@ function assembleQuestionnaire(rootQuestionnaire, totalCanonicals, issues, isRoo
       if (assembled.resourceType === "OperationOutcome") {
         return assembled;
       }
-      subquestionnaire = assembled;
+      subquestionnaires[i] = assembled;
     }
     const prohibitedAttributesOutcome = checkProhibitedAttributes(subquestionnaires);
     if (prohibitedAttributesOutcome) {
